shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/datasource/ShardingSphereDataSource.java

```diff
@@ -113,4 +110,21 @@ private void close(final DataSource dataSource) throws Exception {
     public void close() throws Exception {
         close(getDataSourceMap().keySet());
     }
+    
+    @Override
+    public int getLoginTimeout() throws SQLException {
+        Map<String, DataSource> dataSourceMap = getDataSourceMap();
+        return dataSourceMap.isEmpty() ? 0 : dataSourceMap.values().iterator().next().getLoginTimeout();
+    }
+    
+    @Override
+    public void setLoginTimeout(final int seconds) throws SQLException {
+        for (DataSource each : getDataSourceMap().values()) {
+            each.setLoginTimeout(seconds);
```

Comment:
why not lambda here ?

Reply:
can not use lambda checked exception thrown

Reply:
I see